### PR TITLE
fix(menu-surface): resize the menu's height when the host element's h…

### PR DESCRIPTION
**Pull request by maltja**
…eight changes

fix Lundalogik/crm-feature#1635

## Review:
- [ ] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [ ] Commits have the correct *type* for the changes made
- [ ] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/menu-surface/menu-surface.tsx [hunk 1]
```diff
@@ -61,7 +61,7 @@ export class MenuSurface {
     }
 
     public componentDidUpdate() {
-        this.ensureMenuFitsInViewPort();
+        this.createResizeObserver(this.host);
```
Comment (jgroth) [inline src/components/menu-surface/menu-surface.tsx:64]: Do we have to create a new observer every time the component is updated? :thinking: We should be able to just create one and then remove it when the component is destroyed

Comment (maltja) [reply src/components/menu-surface/menu-surface.tsx:64]: I also thought about that, now the observer is removed in the timeout function. I don't know what is best, only have it when it is resizing(updating) or have it there all the time? but do you think it is better that the observer is always there? 🙂

Comment (jgroth) [reply src/components/menu-surface/menu-surface.tsx:64]: It will allow us to remove the ugly `setTimeout` :sweat_smile:

Comment (maltja) [reply src/components/menu-surface/menu-surface.tsx:64]: haha 😆 I'll do a fix-up then!

**Review by jgroth — DISMISSED** [review]

**Pull request merged**